pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -209,6 +209,23 @@ func (tc *NoExecuteTaintManager) Run(stopCh <-chan struct{}) {
 		}
 	}(stopCh)
 
+	// TODO: Figure out a reasonable number of workers and propagate the
+	// number of workers up making it a paramater of Run() function.
+	workers := 8
+	wg := sync.WaitGroup{}
+	wg.Add(workers)
+	for i := 0; i < workers; i++ {
+		// TODO: With multiple workers running in parallel it may happen that
+		// multiple workers will be processing the same node or pod at the same
+		// time. Can this lead to any races?
```

Comment:
We have discussed this with @gmarek and with this approach we're afaid of "true->false->true' chain of changes to be translated into: first processing "false->true" and only then "true->false".

The solution for that would be to create a number of queues (separate for each worker) and hash nodes into those queues.

I will rewrite this to reflect that.